### DISCUSSION TITLE
How to enrich array of ids in order efficiently

### DISCUSSION THREAD

**Discussion by candidia** [prisma/prisma#14335] [Q&A] [answered]
I have an array of ids and want to get other fields for them and return in the same ordering as the array of ids.

You could do `prisma.model.find` for each but that would be expensive since it would be a new connection for every id.

I am doing `prisma.model.findMany` where `id: { in: array }` but the ordering isn't respected.

Is there some way to achieve this? I'm sure I can figure something out in raw sql but would be convenient to have something built in.

**Comment by TasinIshmam** [comment, marked as answer]
Hi @CSFlorin  

Unfortunately, there's no direct way to achieve this. I would suggest either writing the ordering logic inside your application code (after retrieving the records with `findMany`) or finding an approach with raw SQL.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
